python/runfiles/runfiles.py

```diff
@@ -14,82 +14,117 @@
 
 """Runfiles lookup library for Bazel-built Python binaries and tests.
 
-See README.md for usage instructions.
+See @rules_python//python/runfiles/README.rst for usage instructions.
 """
 import inspect
 import os
 import posixpath
 import sys
+from typing import Dict, Optional, Tuple, Union
 
-if False:
-    # Mypy needs these symbols imported, but since they only exist in python 3.5+,
-    # this import may fail at runtime. Luckily mypy can follow this conditional import.
-    from typing import Callable, Dict, Optional, Tuple, Union
-
-
-def CreateManifestBased(manifest_path):
-    # type: (str) -> _Runfiles
-    return _Runfiles(_ManifestBased(manifest_path))
 
+class _ManifestBased:
```

Comment:
These must be defined earlier to be able to use them as types

Reply:
nit: I thought we could just use a string for the type annotation and I would have assumed that this would have worked irrespectively of where in the file they are written.

Reply:
Isn’t it simpler to reorder them in the long run? I agree it’s annoyingly noisy but I think the outcome will benefit future maintainers and contributors.